### ISSUE TITLE
Request: Allow Login between Zooniverse.org Subdomains

### ISSUE THREAD

**Issue by shaunanoordin** [zooniverse/Panoptes#3123]
## Feature Request

I would like to request the ability for login states to persist between Zooniverse subdomains.

TL;DR: when creating cookies for `_Panoptes_session`, please set cookie's domain to `zooniverse.org` so it works across all subdomains.

### Context and Problem

Students working on the WildCam Darien Lab educational tool first need to **login** at [classroom.zooniverse.org](https://classroom.zooniverse.org/#/wildcam-darien-lab/) so they can see their list of classroom assignments. When they start their classroom assignments, they are taken to WildCam Darien's [Classify page on Zooniverse.org](https://www.zooniverse.org/projects/wildcam/wildcam-darien/classify?workflow=8752&classroom=1&_ga=2.183173990.1927398751.1556897301-1091836498.1556897301) (opened in a new window)

Problem is, their login state is lost when moving from `classroom.zooniverse.org` to `www.zooniverse.org`

![Screen Shot 2019-05-03 at 16 28 48](https://user-images.githubusercontent.com/13952701/57148156-ee429480-6dc0-11e9-8054-444912ae3534.png)
_Student User at WildCam Darien Lab about to start an Assignment. They are logged in. A reminder (yellow) informs them to remember to login at zooniverse.org as well. Clicking on 'Start Assignment' opens a new tab/window to www.zooniverse.org_

![Screen Shot 2019-05-03 at 16 29 04](https://user-images.githubusercontent.com/13952701/57148518-a83a0080-6dc1-11e9-9a04-16bf40393d34.png)
_New tab/window shows the user their Assignment at www.zooniverse.org. However, note that the user is now not logged in, since they switched subdomains._

### Analysis

When the student/user logs in at `classroom.zooniverse.org`, they do so via `panoptes.zooniverse.org`. The `_Panoptes_session` cookie created there is locked to the domain `panoptes.zooniverse.org`

Similarly, if the student/user then logs in at `www.zooniverse.org`, the cookie created there is locked to the domain `www.zooniverse.org` (note that PFE, unlike the various Custom Front Ends, doesn't login via `panoptes.zooniverse.org`)

![Screen Shot 2019-05-03 at 16 40 58](https://user-images.githubusercontent.com/13952701/57148785-49c15200-6dc2-11e9-92f4-d2ee65daf3d1.png)
_Cookie data showing the specific subdomain that's allowed to see it._

### Possible Solution

One possible solution is to specifically set the larger `zooniverse.org` domain when creating Panoptes session cookies, meaning the cookie should be visible to all subdomains of zooniverse.org.

Full solution: https://stackoverflow.com/questions/10402777/share-session-cookies-between-subdomains-in-rails

### Implications

- A "shared login session" change may have an affect on our other projects (i.e. Custom Front Ends) - such as making it impossible for a user to be logged into Account A while at `www.zooniverse.org` while simultaneously logged into Account B while at `antislaverymanuscripts.org` in the same browser...
  - ...though I'm not sure if this is a realistic use case to actually worry about.
- Further details TBD

### Status

Requested

**Comment by srallen** [comment]
This is an issue for Intro2Astro projects as well. We discussed using first party auth for all sub-domained apps back in April and to require all new apps to be sub-domains, so I think this is a good step forward for that goal.

**Comment by srallen** [comment]
@camallen, @wgranger is going to get started on the new text transcription CFE app and we discussed using first party auth and deploying this app to a zooniverse.org subdomain (as we discussed back in April I think for any new apps going forward). I believe this issue would need to be resolved before that would work. Is that correct?

**Comment by rogerhutchings** [comment]
From our chat on Wednesday: we saw the browser security model kick in when trying to access the session cookie on the `frontend.preview.zooniverse.org` subdomain, which meant we couldn't access the cookie in devtools / see it added to the request headers when looking at requests coming in to the rebuild app

**Comment by camallen** [comment]
this has been solved - for oauth clients they need to use the oauth host https://www.zooniverse.org/ once they do that the session cookie will be set for that domain and be shared by browser when visiting our production site at www.zooniverse.org

Technical details: we added CloudFront CDN rules on www.zooniverse.org to proxy to the upstream API host panoptes.zooniverse.org, thus allowing the session cookies to be set for the www.zooniverse.org host.

**Comment by camallen** [comment]
linked to https://github.com/zooniverse/panoptes-javascript-client/pull/126
